_source/_posts/2021-02-26-unity-webgl-playfab-authorization.md

```diff
@@ -0,0 +1,654 @@
+- "User Authorization for @unity3d WebGL and @AzurePlayFab can be easy. Lets step through it in as little as 20 minutes!->"
+- "Friends don't let friends skimp on security! Secure your players identities in your @unity3d WebGL + @AzurePlayFab project in 20 minutes or less."
+Image: blog/unity-webgl-oauth/Webgl_OAuth_Thumb.png
+type: conversion
+---
+
+As game creators, we hold a fundamental responsibility to protect and secure any and all data that our players entrust to us. Historically, security in games has not been high on the priority list because games were enjoyed anonymously and offline. But with our society becoming ever more connected, demand for social gaming experiences and online gaming features has increased exponentially. This has led to the development of cloud-based gaming platforms, such as [PlayFab](https://docs.microsoft.com/en-us/gaming/playfab/what-is-playfab), that handle everything from multiplayer server hosting, leaderboard tracking, virtual economy, data/analytics, and much more. Even games that are mostly considered "offline" are seeing increased demand for some connected functionality. As more games are developed in the cloud (or developed with cloud-based platforms) and track an increasingly large amount of user data, security is becoming one of the highest priorities for the games community.  
+
+{% img blog/unity-webgl-playfab-authorization/p_soti_stuffing.jpeg alt:"SOTI Credential Stuffing" width:"800" %}{: .center-image }
+
+In their 2020 [State of the Internet / Security](https://www.akamai.com/us/en/multimedia/documents/state-of-the-internet/soti-security-gaming-you-cant-solo-security-report-2020.pdf) report, Akamai tracked 100 Billion credential stuffing attacks from June 2018 to June 2020 and found that 10 Billion of these attacks were targeted at gamers. It is not just up to the player or the distribution platform to take security seriously. Players expect that the game companies producing the product that they are trusting their data to will also be keeping them secure.
+
+In [Identity Security for Games in C# with Unity](/blog/2020/08/21/unity-csharp-games-security), we took a look at the extreme basics of storing and authenticating users in a Unity project. That example is a great starting point for the fundamentals, providing a hello world concept within the Unity editor. I want to expand upon how the design can change depending on the build target and how to make use of a player's authorization by passing it into other back-end platforms. To do this, we will be building a [WebGL](https://docs.unity3d.com/Manual/webgl-gettingstarted.html) application in [Unity](https://unity.com/) that authenticates player's and authorizes them to [Azure Playfab](https://playfab.com/); where their player data will be stored.
+
+{% img blog/unity-webgl-playfab-authorization/p_Unity_Playfab_Auth.png alt:"PlayFab SDK" width:"800" %}{: .center-image }
```

Comment:
Suggest adding a sentence above to the effect of: "Here is a conceptual overview of what this will look like:"

Reply:
Stole it